docs/devguide/contributing.rst

```diff
@@ -35,3 +35,76 @@ negative votes, the team will grant commit privileges.
+Configuration files ``.hound.yml`` and ``.jshintrc`` have been created in each 
+repository, so before creating a pull request, you can check code style locally with 
+jshint to find out style violations beforehand.
+
+**Travis CI**
+
+Travis CI runs the unit tests, and then does some potentially ancillary actions.
+The build specifics are detailed in the ``.travis.yml`` file within each repository.
+For finding out basic errors before creating a pull request, you can run unit test
+locally using scripts ``HWIMO-TEST`` within each repository.
+
+**Jenkins**
+
+Jenkins uses the Github Pull Request Builder plugin to monitoring all pull requests
+to perform quality gate tests prior to merge. The gates include running all the unit
+tests, running all dependent project unit tests with the code proposed from the pull 
+request, running an integration "smoke test" to verify basic end to end functionality
+and commenting on the details of test case failure. Jenkins is also able to takes 
+instructions from pull request comments to deal with complex test scenarios. 
+Instructions can be written in pull request description or comments.
+
+The following table show all the Jenkins Instructions and usage:
+
+.. list-table::
+    :widths: 30 50 100 
+    :header-rows: 1
+
+    * - Command
+      - Description
+      - Detailed Usage
+    * - Jenkins: test this please
+      - Trigger one Jenkins test on current pull request.
+      - This command will trigger one Jenkins test whether the pull request has been
+        tested or not, the new test result will cover the previous one. 
+    * - Jenkins: ignore
+      - Avoid running any test in the next Jenkins test.
+      - This command can be used in "work in progress" pull request or one of 
+        "interdependent pull requests" to avoid a Jenkins test that is destined to fail.
+        The commit status of this pull request will be set to "pending" temporally.
+        Jenkins periodically check the repository status so to guarantee to avoid the 
+        unexpected test, it's strongly advised to write this command in pull request description.
+        Otherwise if write in comments, an unexpected test may be triggered in the time slot
+        between pull request creation and writing up comments. 
+    * - Jenkins: depends on pr1_url, pr2_url ...
+      - Trigger one Jenkins test that using the commits of all interdependent pull requests.
+      - RackHD is a multi repositories project, so there are times one new feature leads to 
+        changes on two or more repositories. In such situation neither Jenkins test for single
+        pull request can pass. This command is order to solve this problem. 
+
+        Recommended usage: for interdependent pull requests, first create pull request one by one
+        and write "jenkins: ignore" in description of each pull request until the last one. When
+        create the last pull request write "jenkins: depends on pr1_url, pr2_url ..." in its description.
```

Comment:
Add some words to describe the feature: If the interdependent test success, the test result will be wrote back to all interdependent PRs